spec/hydra/works/models/generic_file_spec.rb

```diff
@@ -31,4 +31,194 @@
 
     it { is_expected.to eq [file1, file2] }
   end
+
+  #blancoj changes.
+  #These are the tests that I added from services/generic_file
+  #add_related_object_spec.rb  get_related_objects_spec.rb  remove_related_object_spec.rb
+   
+  #add_related_object_spec.rb
+  describe 'add related object' do
+
+    let(:subject) { Hydra::Works::GenericFile::Base.new }
+
+    describe 'begin test' do
+
+      context 'with acceptable related objects' do
+        let(:object1) { Hydra::PCDM::Object.create }
+        let(:object2) { Hydra::PCDM::Object.new }
+        let(:generic_work1) { Hydra::Works::GenericWork::Base.create }
+        let(:generic_work2) { Hydra::Works::GenericWork::Base.new }
+        let(:generic_file1) { Hydra::Works::GenericFile::Base.create }
+
+        it 'should add various types of related objects to generic_file' do
+          subject.related_objects << generic_work1
+          subject.related_objects << generic_file1
+          subject.related_objects << object1
+          subject.save
+          subject.reload
+          related_objects = subject.related_objects
+          expect( related_objects.include? generic_work1 ).to be true
+          expect( related_objects.include? generic_file1 ).to be true
+          expect( related_objects.include? object1 ).to be true
+          expect( related_objects.size ).to eq 3
+        end
+
+        context 'with files and generic_files' do
+          let(:file1) { subject.files.build }
+          let(:file2) { subject.files.build }
+          
+          before do
+            subject.save
+            file1.content = "I'm a file"
+            file2.content = "I am too"
+            subject.related_objects << object1
+          end
+
+          it 'should add a related object to a generic_file with files and generic_files' do
+            subject.related_objects << object2 
+            subject.save
+            subject.reload
+            related_objects = subject.related_objects
+            expect( related_objects.include? object1 ).to be true
+            expect( related_objects.include? object2 ).to be true
+            expect( related_objects.size ).to eq 2
+          end
+        end
+      end
+
+      context 'with unacceptable child related objects' do
+        let(:collection1)      { Hydra::Works::Collection.new }
+        let(:pcdm_collection1) { Hydra::PCDM::Collection.new }
+        let(:pcdm_file1)       { Hydra::PCDM::File.new }
+        let(:non_PCDM_object)  { "I'm not a PCDM object" }
+        let(:af_base_object)   { ActiveFedora::Base.new }
+        
+        let(:error_message) { 'child_related_object must be a pcdm object' }
+        
+        it 'should NOT aggregate Hydra::Works::Collection in related objects aggregation' do
+          expect{ Hydra::Works::AddRelatedObjectToGenericFile.call( subject, collection1 ) }.to raise_error(ArgumentError,error_message)
```

Comment:
Please update these tests to use generic_file.related_objects << 

The error raised will be a little different, but you can see the other similar PRs for a template to use: https://github.com/projecthydra-labs/hydra-works/pull/176/files